flycheck.el

```diff
@@ -8128,31 +8128,47 @@ for more information about the custom directories."
                                                  "--print-config" "."))))
     (eq exitcode 0)))
 
+(defun flycheck-parse-eslint (output checker buffer)
+  "Parse ESLint errors/warnings from JSON OUTPUT.
+
+CHECKER and BUFFER denote the CHECKER that returned OUTPUT and
+the BUFFER that was checked respectively.
+
+See URL `https://eslint.org' for more information about ESLint."
+  (let* ((json-object-type 'alist)
+         (json-array-type  'list)
+         (eslint-json-output (car (car (flycheck-parse-json output))))
+         (eslint-errors-list (cdr (assq 'messages eslint-json-output)))
+         (eslint-filename (file-name-nondirectory
+                           (cdr (assq 'filePath eslint-json-output))))
+         errors)
+    (dolist (emessage eslint-errors-list)
+      (let-alist emessage
+        (push (flycheck-error-new-at
+               .line
+               .column
+               (pcase .severity
+                 (2 'error)
+                 (1 'warning)
+                 ;; Default to 'warning for unknown .severity
+                 (_ 'warning))
+               .message
+               :id .ruleId
+               :checker checker
+               :buffer buffer
+               :filename  eslint-filename)
+              errors)))
+    (nreverse errors)))
```

Comment:
I wonder if `mapcar` using a new `flycheck-parse-eslint-1` function would be clearer.

Reply:
@cpitclaudel  I rewrote whole function. Can you review this again?